### PR TITLE
[FLINK-3299] Remove ApplicationID from Environment

### PR DESCRIPTION
**Pull request by uce**
As per discussion in the issue, we decided to remove the the `ApplicationID`.

Replaces the app ID in RocksDB and DB backend with job ID. I've talked to @gyfora about the DB backend and he mentioned a possible problem with savepoints in certain corner cases, but all in all the change should be OK.

### CODE REVIEW

**Comment by tillrohrmann** [comment]
What would that corner case be?

**Comment by gyfora** [comment]
In general the DbStateBackend should not misbehave under the current assumptions. What Ufuk means is that the DbStateBackend does not make any assumption about the job parallelism if using the App id (we could change the job parallelism). This will not hold now but it does not matter as we don't support this feature yet.

File: flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/SavepointCoordinatorTest.java [hunk 1]
```diff
@@ -337,19 +326,14 @@ public void testRollbackStateStoreFailure() throws Exception {
 
 	@Test
 	public void testRollbackUpdatesApplicationID() throws Exception {
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/SavepointCoordinatorTest.java]: test method should renamed

**Comment by zentol** [comment]
should we maybe rename the CompletedCheckpoint class to Savepoint? the naming appears a bit incoonsistent now that the Savepoint class is removed.

**Comment by uce** [comment]
The `CompletedCheckpoint` comes from the `CheckpointCoordinator`. Savepoints are the same thing. Therefore I would keep it as it is.

**Comment by StephanEwen** [comment]
Looks reasonable to me.

**Comment by uce** [comment]
OK, I'm going to merge this then.

**Pull request closed without merging**